sklearn/inspection/_permutation_importance.py

```diff
@@ -32,15 +32,15 @@ def _calculate_permutation_scores(estimator, X, y, col_idx, random_state,
             X_permuted.iloc[:, col_idx] = col
         else:
             X_permuted[:, col_idx] = X_permuted[shuffling_idx, col_idx]
-        feature_score = scorer(estimator, X_permuted, y)
+        feature_score = scorer(estimator, X_permuted, y, sample_weight)
```

Comment:
We can be a little more careful here. If a user passes a scorer that is a callable without `sample_weight`, this would break their code. What about:

```suggestion
		if sample_weight is not None
        	feature_score = scorer(estimator, X_permuted, y, 
        						   sample_weight=sample_weight)
        else:
        	feature_score = scorer(estimator, X_permuted, y)
```

And a small test to make sure passing a callable scorer with the signature:

```
def my_scorer(estimator, X, y):
	return 1
```

does not error.

Reply:
That's a good point. The only issue is that both _permutation_importance_ and __calculate_permutation_scores_ use call, so the suggested correction would have to be implemented in both functions.

An alternative solution is to create a new protected function that do this logic, for example:
`def _scorer_weights_robust(scorer, estimator, X, y, sample_weight):
...    if sample_weight is not None:
...        return scorer(estimator, X, y, sample_weight=sample_weight)
...    return scorer(estimator, X, y)`

then we can use it like this:
`feature_score = _scorer_weights_robust(scorer, estimator, X_permuted, y, sample_weight)`

What do you think?

Reply:
Okay lets create this function to be called in both spots.

Reply:
done